anta/tests/routing/bgp.py

```diff
@@ -447,3 +471,104 @@ def test(self) -> None:
 
         if failures:
             self.result.is_failure(f"Failures: {list(failures.values())}")
+
+
+class VerifyBGPEcmp(AntaTest):
+    """
+    Verifies the installation of BGP Equal-Cost Multipath (ECMP) path in the specified VRF.
+    Expected results:
+        * success: The test will pass if a BGP prefix has ECMP installed and the next hop addresses are the same as the IP route in the specified VRF.
+        * failure: The test will fail if a BGP prefix is not found, ECMP path is not found and the ECMP paths do not match with the IP paths in the specified VRF.
+    """
+
+    name = "VerifyBGPEcmp"
+    description = "Verifies the installation of BGP Equal-Cost Multipath (ECMP) path in the specified VRF."
+    categories = ["routing", "bgp"]
+    commands = [
+        AntaTemplate(template="show bgp ipv4 unicast {prefix} vrf {vrf}"),
+        AntaTemplate(template="show ip route vrf {vrf} {prefix}"),
+    ]
+
+    class Input(AntaTest.Input):
+        """
+        Input parameters for the test.
+        """
+
+        prefixes: List[BgpPrefixes]
+        """List of BGP prefixes"""
+
+        class BgpPrefixes(BaseModel):
+            """
+            This class defines the details of a BGP prefix.
+            """
+
+            prefix: IPv4Network
+            """IPv4 BGP prefix"""
+            vrf: str = "default"
+            """Optional VRF for BGP peer. If not provided, it defaults to `default`."""
+            ecmp_count: int
+            """Maximum number of ECMP paths"""
+
+    def render(self, template: AntaTemplate) -> list[AntaCommand]:
+        """Renders the template with the provided inputs. Returns a list of commands to be executed."""
+
+        return [template.render(prefix=bgp_prefix.prefix, vrf=bgp_prefix.vrf, ecmp_count=bgp_prefix.ecmp_count) for bgp_prefix in self.inputs.prefixes]
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        failures: dict[str, Any] = {}
+        path_data: dict[str, Any] = {}
+        ecmp_install = {}
+
+        # Iterate over command output for different prefixes
+        for command in self.instance_commands:
+            prefix = str(command.params["prefix"])
+            vrf = command.params["vrf"]
+            ecmp_count = command.params["ecmp_count"]
+
+            # Determine if the command is for unicast
+            is_unicast = "unicast" in command.command
```

Comment:
Matching on `is_unicast` can be misleading since we can have other SAFI like `multicast`